controllers/utils/utils.go

```diff
@@ -61,3 +62,14 @@ func MinRequeueDuration(old, new *time.Duration) *time.Duration {
 	}
 	return old
 }
+
+func GetAllRemediationTemplates(nhc *v1alpha1.NodeHealthCheck) []*v1.ObjectReference {
+	refs := make([]*v1.ObjectReference, 1)
+	if nhc.Spec.RemediationTemplate != nil {
+		refs = append(refs, nhc.Spec.RemediationTemplate)
```

Comment:
Nit: just for clarity consider a return statement here since RemediationTemplate & EscalationRemediation are mutually exclusive

Reply:
done
well, refactored 😉